### PR TITLE
Update recipe arg overwrite to support list types

### PR DESCRIPTION
**Pull request by KSGulin**
Currently, utilizing `--recipe_args` with a recipe that has a [2, 4] mask or a list-valued entry for any of the fields, will result in an error. This is because the recipe update logic makes the assumption that any list-valued entry is a list of modifiers (represented by dictionaries). This PR fixes this with a minor logic change.

**Testing**
```bash
sparseml.yolov5.train --cfg models_v5.0/yolov5l.yaml --recipe-args '{\"quantization_start_epoch\":1}' --recipe recipes/test_recipe.md
```
Where `recipes/test_recipe.md` is

```yaml
---
# General Hyperparams
init_lr: 0.01
weights_warmup_lr: 0

# Pruning Hyperparams
pruning_update_frequency: 0.5
mask_type: [1, 4]

# Quantization Hyperparams
quantization_start_epoch: 120

training_modifiers:
  - !LearningRateFunctionModifier
    start_epoch: 0
    end_epoch: 3
    lr_func: linear
    init_lr: eval(weights_warmup_lr)
    final_lr: eval(init_lr)
    param_groups: [0, 1]
        
quantization_modifiers:
  - !QuantizationModifier
    start_epoch: eval(quantization_start_epoch)
---
```

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]

File: src/sparseml/optim/helpers.py [hunk 1]
```diff
@@ -255,7 +255,9 @@ def _update_recipe_variable(var_name: str, var_value, container):
     key_found = False
 
     for key, value in container.items():
-        if not isinstance(value, list):
+        if not isinstance(value, list) or not all(
+            [isinstance(val, dict) for val in value]
```
Comment (rahul-tuli) [inline src/sparseml/optim/helpers.py:259]:
```suggestion
            isinstance(val, dict) for val in value
```

**Review by rahul-tuli — DISMISSED** [review]
Go `Nittany Lions`

File: src/sparseml/optim/helpers.py [hunk 2]
```diff
@@ -255,7 +255,9 @@ def _update_recipe_variable(var_name: str, var_value, container):
     key_found = False
 
     for key, value in container.items():
-        if not isinstance(value, list):
+        if not isinstance(value, list) or not all(
```
Comment (corey-nm) [inline src/sparseml/optim/helpers.py:258]: Can we add some small unit tests for this just to be sure that this functionality isn't lost in the future?

Comment (KSGulin) [reply src/sparseml/optim/helpers.py:258]: Added coverage to existing unit tests and confirmed it fails on main without this PR

**Review by bfineran — APPROVED** [review]

**Review by corey-nm — APPROVED** [review]

**Pull request merged**